File: automap/src/main/java/com/userjhansen/automap/Maps/InsideOne.java
package com.userjhansen.automap.Maps;

import com.acmerobotics.roadrunner.geometry.Pose2d;
import com.userjhansen.automap.AutoPart;
import com.userjhansen.automap.PartType;

public class InsideOne implements Map {
    public Pose2d startPosition = new Pose2d(-35, -60, -Math.PI / 2);

    public static Pose2d depositPosition = new Pose2d(-60, -50, -(Math.PI / 16) + (Math.PI/2));

    public static AutoPart[] startParts = {
            new AutoPart(PartType.STRAFE, new Pose2d(-9, -32,0)),
            new AutoPart(PartType.ACTION, 0),

            new AutoPart(PartType.SPLINE_TO, new Pose2d(-48, -40, Math.PI/2), Math.PI),
            new AutoPart(PartType.STRAFE_TO, new Pose2d(-48, -35, Math.PI/2)),
            new AutoPart(PartType.ACTION, 1),
            new AutoPart(PartType.STRAFE_TO, depositPosition),
            new AutoPart(PartType.ACTION, 2),

            new AutoPart(PartType.STRAFE_TO, new Pose2d(-59, -35, Math.PI/2)),
            new AutoPart(PartType.ACTION, 1),
            new AutoPart(PartType.STRAFE_TO, depositPosition),
            new AutoPart(PartType.ACTION, 2),

            new AutoPart(PartType.STRAFE_TO, new Pose2d(-54, -24, -Math.PI)),
            new AutoPart(PartType.STRAFE, new Pose2d(-57, -24, -Math.PI)),
            new AutoPart(PartType.ACTION, 1),
            new AutoPart(PartType.STRAFE_TO, depositPosition),
            new AutoPart(PartType.ACTION, 2),

            new AutoPart(PartType.STRAFE_TO, new Pose2d(-50, -12, 0)),
            new AutoPart(PartType.STRAFE_TO, new Pose2d(-22, -12, 0)),
    };

    @Override
    public Pose2d getStartPosition() {
        return startPosition;
    }

    @Override
    public AutoPart[] getStartParts() {
        return startParts;
    }
}

File: automap/src/main/java/com/userjhansen/automap/Maps/InsideOne.kt
package com.userjhansen.automap.Maps

import com.acmerobotics.roadrunner.Pose2d
import com.userjhansen.automap.AutoPart
import com.userjhansen.automap.PartType

class InsideOne : Map {
    override val startPosition = Pose2d(15.0, -62.0, Math.PI / 2)

    override val specimenPosition = Pose2d(2.0, -30.0, Math.PI / 2)
    override val specimenPosition2 = Pose2d(1.0, -29.0, Math.PI / 2)
    override val depositPosition = Pose2d(-46.0, -58.0, Math.PI / 16)
    override val collectPosition = Pose2d(42.0, -60.0, Math.PI / 2)

    override val intakeParts = arrayOf(
        arrayOf(
            AutoPart(PartType.STRAFE, Pose2d(-48.0, -40.0, Math.PI / 2)),
            AutoPart(PartType.STRAFE_TO, Pose2d(-48.0, -35.0, Math.PI / 2)),
        ),
        arrayOf(
            AutoPart(PartType.STRAFE_TO, Pose2d(-59.0, -35.0, Math.PI / 2.0)),
        ),
        arrayOf(
            AutoPart(PartType.STRAFE_TO, Pose2d(-54.0, -24.0, -Math.PI)),
            AutoPart(PartType.STRAFE, Pose2d(-57.0, -24.0, -Math.PI)),
        ),
    )

    override val depositParts = arrayOf(
        AutoPart(PartType.STRAFE_TO, depositPosition),
    )

    override val parkParts = arrayOf(
        AutoPart(PartType.STRAFE_TO, Pose2d(59.0, -39.0, 0.0)),
        AutoPart(PartType.STRAFE_TO, Pose2d(59.0, -60.0, 0.0)),
    )

    override val specimenParts = arrayOf(
//        Specimen Collecting
        AutoPart(PartType.STRAFE_TO, Pose2d(42.0, -60.0, Math.PI / 2)),
    )
}